packages/office-ui-fabric-react/src/components/Slider/Slider.styles.ts

```diff
@@ -1,30 +1,251 @@
 import { ISliderStyleProps, ISliderStyles } from './Slider.types';
-import { getGlobalClassNames } from '../../Styling';
+import { getGlobalClassNames, HighContrastSelector, AnimationVariables, getFocusStyle } from '../../Styling';
+import { getRTL } from '@uifabric/utilities';
 
 const GlobalClassNames = {
   root: 'ms-Slider',
+  enabled: 'ms-Slider-enabled',
+  disabled: 'ms-Slider-disabled',
+  row: 'ms-Slider-row',
+  column: 'ms-Slider-column',
   container: 'ms-Slider-container',
   slideBox: 'ms-Slider-slideBox',
   line: 'ms-Slider-line',
   thumb: 'ms-Slider-thumb',
   activeSection: 'ms-Slider-active',
   inactiveSection: 'ms-Slider-inactive',
-  valueLabel: 'ms-Slider-value'
+  valueLabel: 'ms-Slider-value',
+  showValue: 'ms-Slider-showValue',
+  showTransitions: 'ms-Slider-showTransitions'
 };
 
 export const getStyles = (props: ISliderStyleProps): ISliderStyles => {
-  const { className, theme } = props;
+  const { className, titleLabel, theme } = props;
   const classNames = getGlobalClassNames(GlobalClassNames, theme);
 
+  const slideBoxActiveSectionStyles = props.rootIsEnabled && {
+    backgroundColor: theme.palette.themePrimary,
+    selectors: {
+      [HighContrastSelector]: {
+        backgroundColor: 'Highlight'
+      }
+    }
+  };
+
+  const slideBoxInactiveSectionStyles = props.rootIsEnabled && {
+    backgroundColor: theme.palette.themeLight,
+    selectors: {
+      [HighContrastSelector]: {
+        borderColor: 'Highlight'
+      }
+    }
+  };
+
+  const slideBoxActiveThumbStyles = props.rootIsEnabled && {
+    border: `2px solid ${theme.palette.themePrimary}`,
+    selectors: {
+      [HighContrastSelector]: {
+        borderColor: 'Highlight'
+      }
+    }
+  };
+
   return {
-    root: [classNames.root, {}, className],
-    container: [classNames.container, {}],
-    slideBox: [classNames.slideBox, {}],
-    line: [classNames.line, {}],
-    thumb: [classNames.thumb, {}],
+    root: [
+      classNames.root,
+      {
+        userSelect: 'none'
+      },
+      props.rootIsVertical && {
+        marginRight: 8
+      },
+      className,
```

Comment:
Provided className should be listed last to make sure it gets priority in styling... also, is there a test for a provided className?

Reply:
done